[ONNX] Update docs on symbolic registration

justinchuby opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #85350
* #85349
* #85312
* #85165
* #84776
* #84728
* __->__ #85290
* #84448

- Move inline instructions on editing symbolic functions to the README
- Add a line on using the symbolic function registration decorator.

justinchuby committed:
[ONNX] Update docs on symbolic registration

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Update docs on symbolic registration"


- Move inline instructions on editing symbolic functions to the README
- Add a line on using the symbolic function registration decorator.

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Update docs on symbolic registration"


- Move inline instructions on editing symbolic functions to the README
- Add a line on using the symbolic function registration decorator.

[ghstack-poisoned]

justinchuby commented:
@BowenBao PTAL

justinchuby committed:
Update on "[ONNX] Update docs on symbolic registration"


- Move inline instructions on editing symbolic functions to the README
- Add a line on using the symbolic function registration decorator.

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Update docs on symbolic registration"


- Move inline instructions on editing symbolic functions to the README
- Add a line on using the symbolic function registration decorator.

[ghstack-poisoned]

BowenBao commented on torch/onnx/README.md:
```diff
@@ -23,3 +24,75 @@ symbolic_opset9.py.
+    Python, which may not match the tensor it is being added to.  PyTorch
+    will do implicit conversions on scalars; however, ONNX will not, so
+    we must do the conversion ourselves.  This is what `symbolic_helper._if_scalar_type_as()`
+    does.
```
Noticed this is a little outdated, best add
```suggestion
    and `_jit_pass_onnx_scalar_type_analysis` does.
```

justinchuby replied:
Done

justinchuby committed:
Update on "[ONNX] Update docs on symbolic registration"


- Move inline instructions on editing symbolic functions to the README
- Add a line on using the symbolic function registration decorator.

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Update docs on symbolic registration"


- Move inline instructions on editing symbolic functions to the README
- Add a line on using the symbolic function registration decorator.

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Update docs on symbolic registration"


- Move inline instructions on editing symbolic functions to the README
- Add a line on using the symbolic function registration decorator.

[ghstack-poisoned]

justinchuby commented:
@pytorchbot merge
